openapi/src/main/java/io/helidon/openapi/ImplTypeDescription.java

```diff
@@ -0,0 +1,168 @@
+/*
+ * Copyright (c) 2020 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package io.helidon.openapi;
+
+import java.util.Set;
+
+import org.yaml.snakeyaml.TypeDescription;
+import org.yaml.snakeyaml.introspector.Property;
+import org.yaml.snakeyaml.introspector.PropertySubstitute;
+import org.yaml.snakeyaml.introspector.PropertyUtils;
+import org.yaml.snakeyaml.nodes.Node;
+import org.yaml.snakeyaml.nodes.Tag;
+
+/**
+ * Version of {@code TypeDescription} used for implementation classes (rather than the corresponding interfaces) which delegate
+ * most method invocations to the {@code TypeDescription} for the related interface.
+ */
+class ImplTypeDescription extends TypeDescription {
```

Comment:
Layman reviewer note: surprised me to see `Impl` as a prefix; was expecting a suffix here.  This is true even when I read your class comment.  I'm not sure what would be a better name.  Maybe `DelegatingTypeDescription`?

Reply:
I think an `Impl` suffix might be problematic because this class is a `TypeDescription` for implementation classes (from SmallRye) rather than for interfaces (from MP OpenAPI). Plus `TypeDescription` is itself a class, so I'm not sure about the expectation of seeing an "implementation" of another class.

The apparent clarity of `DelegatingTypeDescription` might actually be a little misleading, because, although it delegates every method to `delegate.something`, the `getType` method_does not_ delegate to `delegate.getType()` but to `delegate.impl()`. This is what makes this an `ImplTypeDescription`.

I will make this more explicit in the class comments.